### PR TITLE
TRUNK-5212 Add test case for ProgramWorkFlowService.getProgramsByConcept 

### PR DESCRIPTION
**Pull request by dilekamadushan**
https://issues.openmrs.org/browse/TRUNK-5212?jql=assignee%20in%20(madushan)
## Description of what I changed
 added test case to get program by concept in ProgramWorkFlowServiceTest 

## Checklist: I completed these to help reviewers :)

- [ x] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

- [x ] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

- [ x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

- [x ] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

- [x ] All new and existing **tests passed**.

- [ x] My pull request is **based on the latest changes** of the master branch.

### CODE REVIEW

File: api/pom.xml [hunk 1]
```diff
@@ -97,6 +97,14 @@
          	</exclusion>
          </exclusions>
       </dependency>
+      <dependency>
+         <groupId>ca.uhn.hapi</groupId>
+         <artifactId>hapi-structures-v23</artifactId>
```
Comment (dkayiwa) [inline api/pom.xml]: Are these dependencies necessary for adding the tests?

File: pom.xml [hunk 2]
```diff
@@ -336,6 +336,16 @@
 				<artifactId>hapi-base</artifactId>
 				<version>2.0</version>
 			</dependency>
+			<dependency>
+				<groupId>ca.uhn.hapi</groupId>
+				<artifactId>hapi-structures-v23</artifactId>
+				<version>2.0</version>
```
Comment (dkayiwa) [inline pom.xml]: Are these dependencies necessary for adding the tests?

Comment (dilekamadushan) [reply pom.xml]: @dkayiwa I will make the neccessary changes. Thanks. Please give feedback on #2479

**Comment by dilekamadushan** [comment]
@dkayiwa I modified my PR on your latest reviews. Feel free to review. Thanks

**Comment by dilekamadushan** [comment]
@teleivo Feel free to review on latest changes

File: api/src/test/java/org/openmrs/api/ProgramWorkflowServiceTest.java [hunk 3]
```diff
@@ -67,6 +67,9 @@ public void runBeforeEachTest() {
 			encounterService = Context.getEncounterService();
 			cs = Context.getConceptService();
 		}
+
+        // taking program workflow service from the spring application context object
```
Comment (dkayiwa) [inline api/src/test/java/org/openmrs/api/ProgramWorkflowServiceTest.java]: As for the above comment, take a look at this: https://dzone.com/articles/5-best-practices-commenting

Comment (dilekamadushan) [reply api/src/test/java/org/openmrs/api/ProgramWorkflowServiceTest.java]: @dkayiwa thanks I will make the necessary changes.

**Comment by dilekamadushan** [comment]
@dkayiwa I resolved the issue. Thanks

**Comment by teleivo** [comment]
It seems to me that this issue is a duplicate of issue worked on here #2536
I've asked the creator of the issue for more detail since from the issue type its a bug but the issue description just task about adding a test which would make the issue a duplicate. I am closing this for now while waiting on https://issues.openmrs.org/browse/TRUNK-5212

please continue your work on #2536

**Comment by dilekamadushan** [comment]
@teleivo What about the test case added in this PR. Should I add this method to #2536

**Comment by teleivo** [comment]
no, first start small on the other PR with the instructions I posted there
